src/st/rwrp/8DF0.c

```diff
@@ -7,9 +7,94 @@
 #include "game.h"
 #include "rwrp.h"
 
-INCLUDE_ASM("st/rwrp/nonmatchings/8DF0", func_80188DF0);
+extern ObjInit2 D_801804E0[];
 
-INCLUDE_ASM("st/rwrp/nonmatchings/8DF0", func_80188ED0);
+void func_80188DF0(Entity* arg0) {
+    ObjInit2* temp_s1;
+    u32 temp_v0;
+
+    temp_s1 = &D_801804E0[arg0->params];
+    if (arg0->step == 0) {
+        InitializeEntity(g_eInitGeneric2);
+        arg0->animSet = (s16)temp_s1->animSet;
+        arg0->zPriority = temp_s1->zPriority;
+        arg0->unk5A = (u16)temp_s1->unk4.s;
+        arg0->palette = temp_s1->palette;
+        arg0->drawFlags = temp_s1->drawFlags;
+        arg0->drawMode = temp_s1->drawMode;
+        temp_v0 = temp_s1->unkC;
+        if (temp_v0 != 0) {
+            arg0->flags = (s32)temp_v0;
+        }
+        if (arg0->params == 1) {
+            arg0->rotY = 0x200;
+            arg0->rotX = 0x200;
+        }
+    }
+    AnimateEntity((s32)temp_s1->unk10, arg0);
+}
+
+extern u16 D_80180538[];
+extern u16 D_80180488[];
+extern u8 D_80180530[];
+extern u8 D_80180528[];
+
+void EntityUnkId12(Entity* entity) {
+    s32 ret;
+    u16* temp_v0_2;
+    u16 temp_s1 = entity->params;
+    u16 phi_v1;
+    u16 unk;
+
+    entity->unk6D[0] = 0;
+
+    if (entity->step != 0) {
+        switch (temp_s1) {
+        case 4:
+        case 5:
+            if (g_Tilemap.x != 0) {
+                return;
+            }
+            break;
+
+        case 6:
+            if (g_pads->pressed & PAD_TRIANGLE) {
+                g_Tilemap.x = 0;
+                g_Tilemap.width = 1280;
+                entity->step++;
+                return;
+            }
+            break;
+        }
+
+        if (entity->unk44 != 0) {
+            ret = GetSideToPlayer();
+            phi_v1 = entity->ext.generic.unk7C.s;
```

Comment:
There's a bunch of entity->ext.generic, we're trying to get rid of these https://github.com/Xeeynamo/sotn-decomp/issues/750